src/components/Card.js

```diff
@@ -1,41 +1,99 @@
+import Api from "../components/Api.js";
+import DeleteCardForm from "./DeleteCardForm.js";
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "b685d3e0-616a-4dae-bc5b-53892a4f7953",
+    "Content-Type": "application/json",
+  },
+});
+
 export default class Card {
-  constructor(cardData, cardSelector, handleCardClick) {
+  constructor(
+    cardData,
+    cardSelector,
+    handleCardClick,
+    handleCardDeleteFunctionInIndexComponent
+  ) {
     this._cardData = cardData;
+    this._id = cardData._id;
     this._name = cardData.name;
     this._link = cardData.link;
+    this._owner = cardData.owner._id;
+    this._cardId = cardData._id;
+    this._likes = cardData.likes;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleCardDeleteFunctionInIndexComponent =
+      handleCardDeleteFunctionInIndexComponent;
   }
 
-  _handleFavIcon = () => {
-    this._favIconElement.classList.toggle("card__fav-icon-selected");
+  _handleFavIconClick = () => {
+    //variable to register if current user liked the image or not
+    const currentUserLike = this._likes.find((user) => {
+      return user._id === "f50447686616d1fa985ca0e1";
```

Comment:
`f50447686616d1fa985ca0e1` is a hardcode


`_id` of the user should be received from the server only 1 time. You need to store it in a variable `userId` or in class `UserInfo` and use it when you create a card passing it to the constructor of `Card`